Formula/spack.rb

```diff
@@ -0,0 +1,21 @@
+class Spack < Formula
+  desc "Package manager that builds multiple versions and configurations of software"
+  homepage "https://spack.io"
+  url "https://github.com/spack/spack/releases/download/v0.16.0/spack-0.16.0.tar.gz"
```

Comment:
```suggestion
  url "https://github.com/spack/spack/archive/v0.16.0.tar.gz"
```
I think we usually prefer the source code than the actual release tarball.